### PR TITLE
PostgreSQL Plugin

### PR DESCRIPTION
**Pull request by jlsuttles**
Adds the ability to save cypress log messages to a postgresql database.

### CODE REVIEW

File: plugin/postgresql_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+	n.CheckMock(&db.Mock)
+
+	n.It("sets up a db", func() {
+		var p PostgreSQL
+		p.Init(&db)
+
+		db.On("Ping").Return(nil)
+		db.On("Exec", cCreateTable, []interface{}{}).Return(mock.Anything, nil)
+		db.On("Exec", cEnableHstore, []interface{}{}).Return(mock.Anything, nil)
```
Comment (jlsuttles) [inline plugin/postgresql_test.go]: @evanphx This mock is giving me trouble. Ideas?

```
panic: 

mock: Unexpected Method Call
-----------------------------

Exec(string,[]interface {})
        0: 
CREATE EXTENSION hstore

        1: []

The closest call I have is: 

Exec(string,[]interface {})
        0: 
CREATE EXTENSION hstore

        1: []
```

Comment (evanphx) [inline plugin/postgresql_test.go]: Ug, mocks. I wonder if this is the difference between an empty slice and a nil slice. Try changing 28 to `[]interface{}(nil)` or just remove that param all together.

Comment (jlsuttles) [inline plugin/postgresql_test.go]: Removing that param all together did not work, but using `[]interface{}(nil)` solves that problem. Yay :confetti_ball:

**Comment by evanphx** [comment]
Woop!

**Pull request merged**